interface_notes.md

```diff
@@ -0,0 +1,387 @@
+  warn(msg, DeprecationWarning, stacklevel=1)
+/Users/jsolbrig/NRL/geoips_packages/geoips/geoips/dev/alg.py:202: DeprecationWarning: Algorithm attribute "alg_func_type", used in visir.Night_Vis_IR, is deprecated and will be removed in a future release. Please replace all occurrences with "func_type".
+  warn(msg, DeprecationWarning, stacklevel=1)
+/Users/jsolbrig/NRL/geoips_packages/geoips/geoips/dev/alg.py:202: DeprecationWarning: Algorithm attribute "alg_func_type", used in visir.Night_Vis_IR_GeoIPS1, is deprecated and will be removed in a future release. Please replace all occurrences with "func_type".
+  warn(msg, DeprecationWarning, stacklevel=1)
+[26, 19, 59]
+name                       | type                | description                                                
+--------------------------------------------------------------------------------------------------------------
+pmw_tb.pmw_37pct           | list_numpy_to_numpy | Passive Microwave 37 MHz Polarization Corrected Temperature
+pmw_tb.pmw_89pct           | list_numpy_to_numpy |                                                            
+pmw_tb.pmw_color37         | list_numpy_to_numpy |                                                            
+pmw_tb.pmw_color89         | list_numpy_to_numpy |                                                            
+sfc_winds.windbarbs        | list_numpy_to_numpy |                                                            
+single_channel             | list_numpy_to_numpy |                                                            
+visir.Night_Vis            | list_numpy_to_numpy |                                                            
+visir.Night_Vis_GeoIPS1    | list_numpy_to_numpy |                                                            
+visir.Night_Vis_IR         | list_numpy_to_numpy |                                                            
+visir.Night_Vis_IR_GeoIPS1 | list_numpy_to_numpy |                                                            
+```
+
+# Important to do for review
+- ~~Ensure that the output of alg.get_list(by_type=True) fits what it was before. Something doesn't look right.~~
+  - I think I have this fixed. Still worth checking, but it doesn't look like this interface was working correctly anyway!
+
+# Questions
+- Need to figure out what we want to call `Boundaries` and `Gridlines` which are both yaml-config based. These likely
+  shouldn't be combined into a single class because they are sometimes mixed and matched.
+  - Maybe have a MapInterface module that defines which Boundaries and Gridlines you want to use.  
+- It looks like the `output_config` interface implements `get_output_format` and `get_output_format...`. These need to
+  be updated to work better with the new system and have appropriate names.
+- What is the difference between `geoips/dev/utils.py` and `geoips/geoips_utils.py`? It seems like a lot of what is in
```

Comment:
@jsolbrig Probably can move things from dev/utils.py to geoips_utils.py (and should do that when we get rid of the rest of the YAML based interfaces in dev)